### PR TITLE
[5.3] Explicit route model binding specific to a route group

### PR DESCRIPTION
**Pull request by hasyimibhar**
Refer to issue #12334. This PR allows defining explicit binding specific to a route group:

``` php
$router->group(['namespace' => 'Foo'], function ($router, $group) {
    $group->bind('user', function($value) {
        // ...
    });

    $router->get('users/{user}', 'FooController@show');
    // ...
});
```

Do tell me if this should go to `5.2` branch instead, since it's not breaking (I think).

### CODE REVIEW

File: src/Illuminate/Routing/Route.php [hunk 1]
```diff
@@ -105,13 +112,15 @@ class Route
      * @param  array   $methods
      * @param  string  $uri
      * @param  \Closure|array  $action
+     * @param  \Illuminate\Routing\RouteGroup  $group
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Route.php]: missing |null

File: src/Illuminate/Routing/Route.php [hunk 2]
```diff
@@ -393,6 +402,16 @@ public function parameterNames()
     }
 
     /**
+     * Get the group the route belongs to.
+     *
+     * @return \Illuminate\Routing\RouteGroup
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Route.php]: |null

File: src/Illuminate/Routing/Route.php [hunk 3]
```diff
@@ -44,6 +44,13 @@ class Route
     protected $action;
 
     /**
+     * The route group it belongs to.
+     *
+     * @var /Illuminate/Routing/RouteGroup
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Route.php]: |null

File: src/Illuminate/Routing/RouteGroup.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+     * The registered route value binders.
+     *
+     * @var array
+     */
+    protected $binders = [];
+
+    /**
+     * Create a new RouteGroup instance.
+     *
+     * @param /Illuminate/Routing/RouteGroup  $parent
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/RouteGroup.php]: missing |null everywhere

File: tests/Routing/RoutingRouteTest.php [hunk 5]
```diff
@@ -428,6 +428,139 @@ public function testRouteClassMethodBinding()
...
+        });
+
+        $router->get('boo/{baz}', function ($name) { return $name; });
+
+        $this->assertEquals('TAYLOROMG', $router->dispatch(Request::create('foo/bar/whatever', 'GET'))->getContent());
+        $this->assertEquals('whatever', $router->dispatch(Request::create('blah/whatever', 'GET'))->getContent());
+        $this->assertEquals('whatever', $router->dispatch(Request::create('boo/whatever', 'GET'))->getContent());
+
+        /*
+         * Alternative syntax
```
Comment (GrahamCampbell) [inline tests/Routing/RoutingRouteTest.php]: can you break each of these commented sections in to seperate tests please?

Comment (hasyimibhar) [inline tests/Routing/RoutingRouteTest.php]: Sure :+1: .

**Comment by taylorotwell** [comment]
No plans to bring this in at this time.

**Pull request closed without merging**